### ISSUE TITLE
IE Driver Server Crashes When Trying to Get Cookies

### ISSUE THREAD

**Issue by bjharo** [SeleniumHQ/selenium#5054]
## Meta -
OS:  
Windows 10 Pro (64 bit)
Version 1709
Build: 16299.19

Selenium Version:  
3.6.0

Browser:  
Internet Explorer 

Browser Version:  
11.15.16299.0

## Expected Behavior - 
IE Driver Server successfully gets the cookies for the current session and a specific cookie can be retrieved.

## Actual Behavior -
While the specific line of code to get the cookies from the current browser session appears to work, upon trying to get a specific cookie from said collection, the IE Driver Server executable crashes.  Literally, it just disappears from Task Manager at this point.

## Steps to reproduce -

I have multiple tests where I need to download files from the web app I'm testing.  Due to the nature of how these files are handled within the app, I actually have to download the files outside of the browser.  To do this, I'm using .NET WebClient and I retrieve the session cookie from the current browser instance and use that in my WebClient request.  I've been doing this for a while now and it has not been a problem on any of the three major browser I test (Chrome, Firefox, IE).    Just today on my laptop, IE started giving me problems and I tracked it down to this point in my code:

```
var cookies = WebDriver.Manage().Cookies;
var sessionId = cookies.GetCookieNamed(SESSION_COOKIE_NAME).Value;
```

The first line appears to execute fine.  Upon executing the second, the IE Driver Server executable crashes.  Here are the relevant lines from the IE Driver Server log:

```
T 2017-11-14 15:09:05:339 server.cc(391) Entering Server::SendResponseToClient
T 2017-11-14 15:09:05:339 response.cc(30) Entering Response::Deserialize
T 2017-11-14 15:09:05:339 server.cc(445) Entering Server::SendHttpOk
T 2017-11-14 15:09:38:283 server.cc(174) Entering Server::ProcessRequest
T 2017-11-14 15:09:38:283 server.cc(183) Process request with: URI: /session/8cd373e1-4e1e-4a32-a716-257ea34a8222/cookie HTTP verb: GET
body: {}
T 2017-11-14 15:09:38:283 server.cc(272) Entering Server::DispatchCommand
T 2017-11-14 15:09:38:283 server.cc(583) Entering Server::LookupCommand
D 2017-11-14 15:09:38:283 server.cc(281) Command: GET /session/8cd373e1-4e1e-4a32-a716-257ea34a8222/cookie {}
T 2017-11-14 15:09:38:283 server.cc(378) Entering Server::LookupSession
T 2017-11-14 15:09:38:283 IESession.cpp(191) Entering IESession::ExecuteCommand
T 2017-11-14 15:09:38:283 IECommandExecutor.cpp(122) Entering IECommandExecutor::OnSetCommand
T 2017-11-14 15:09:38:283 command.cc(31) Entering Command::Deserialize
D 2017-11-14 15:09:38:283 command.cc(36) Raw JSON command: { "name" : "getAllCookies", "locator" : { "sessionid" : "8cd373e1-4e1e-4a32-a716-257ea34a8222" }, "parameters" : {} }
T 2017-11-14 15:09:38:283 IECommandExecutor.cpp(133) Entering IECommandExecutor::OnExecCommand
T 2017-11-14 15:09:38:283 IECommandExecutor.cpp(454) Entering IECommandExecutor::DispatchCommand
T 2017-11-14 15:09:38:283 IECommandExecutor.cpp(577) Entering IECommandExecutor::GetCurrentBrowser
T 2017-11-14 15:09:38:283 IECommandExecutor.cpp(583) Entering IECommandExecutor::GetManagedBrowser
T 2017-11-14 15:09:38:283 IECommandExecutor.cpp(529) Entering IECommandExecutor::IsAlertActive
T 2017-11-14 15:09:38:283 IESession.cpp(212) Beginning wait for response length to be not zero
T 2017-11-14 15:09:38:283 Browser.cpp(700) Entering Browser::GetActiveDialogWindowHandle
T 2017-11-14 15:09:38:283 Browser.cpp(206) Entering Browser::GetContentWindowHandle
D 2017-11-14 15:09:38:283 IECommandExecutor.cpp(543) No alert handle is found
T 2017-11-14 15:09:38:283 IECommandExecutor.cpp(577) Entering IECommandExecutor::GetCurrentBrowser
T 2017-11-14 15:09:38:283 IECommandExecutor.cpp(583) Entering IECommandExecutor::GetManagedBrowser
T 2017-11-14 15:09:38:283 DocumentHost.cpp(73) Entering DocumentHost::GetCurrentUrl
T 2017-11-14 15:09:38:283 Browser.cpp(126) Entering Browser::GetDocument
I 2017-11-14 15:09:38:283 Browser.cpp(130) No child frame focus. Focus is on top-level frame
T 2017-11-14 15:09:38:291 Browser.cpp(613) Entering Browser::GetDocumentFromWindow
T 2017-11-14 15:09:38:297 CookieManager.cpp(86) Entering CookieManager::GetCookies
T 2017-11-14 15:09:38:298 HookProcessor.cpp(104) Entering HookProcessor::Initialize
T 2017-11-14 15:09:38:298 HookProcessor.cpp(204) Entering HookProcessor::CreateReturnPipe
D 2017-11-14 15:09:38:298 HookProcessor.cpp(239) Created named pipe \\.\pipe\IEDriverPipe9956
T 2017-11-14 15:09:38:298 HookProcessor.cpp(167) Entering HookProcessor::InstallWindowsHook
T 2017-11-14 15:09:38:298 CookieManager.cpp(369) Entering CookieManager::SendGetCookieMessage
T 2017-11-14 15:09:38:298 HookProcessor.cpp(261) Entering HookProcessor::PushData
T 2017-11-14 15:09:38:301 HookProcessor.cpp(287) Entering HookProcessor::PullData
D 2017-11-14 15:09:38:301 HookProcessor.cpp(293) Waiting for connection from browser via named pipe
T 2017-11-14 15:09:38:301 CookieManager.cpp(398) Entering CookieManager::ThreadProc
D 2017-11-14 15:09:38:301 HookProcessor.cpp(302) Connection from browser established via named pipe
T 2017-11-14 15:09:38:301 CookieManager.cpp(349) Entering CookieManager::ParseCookieString
T 2017-11-14 15:09:38:302 CookieManager.cpp(369) Entering CookieManager::SendGetCookieMessage
T 2017-11-14 15:09:38:302 HookProcessor.cpp(261) Entering HookProcessor::PushData
T 2017-11-14 15:09:38:303 HookProcessor.cpp(287) Entering HookProcessor::PullData
D 2017-11-14 15:09:38:303 HookProcessor.cpp(293) Waiting for connection from browser via named pipe
T 2017-11-14 15:09:38:303 CookieManager.cpp(398) Entering CookieManager::ThreadProc
D 2017-11-14 15:09:38:306 HookProcessor.cpp(302) Connection from browser established via named pipe
T 2017-11-14 15:09:38:306 CookieManager.cpp(349) Entering CookieManager::ParseCookieString
T 2017-11-14 15:09:38:306 CookieManager.cpp(369) Entering CookieManager::SendGetCookieMessage
T 2017-11-14 15:09:38:306 HookProcessor.cpp(261) Entering HookProcessor::PushData
T 2017-11-14 15:09:38:307 HookProcessor.cpp(287) Entering HookProcessor::PullData
D 2017-11-14 15:09:38:307 HookProcessor.cpp(293) Waiting for connection from browser via named pipe
T 2017-11-14 15:09:38:307 CookieManager.cpp(398) Entering CookieManager::ThreadProc
D 2017-11-14 15:09:38:307 HookProcessor.cpp(302) Connection from browser established via named pipe
T 2017-11-14 15:09:38:307 CookieManager.cpp(349) Entering CookieManager::ParseCookieString
T 2017-11-14 15:09:38:307 CookieManager.cpp(369) Entering CookieManager::SendGetCookieMessage
T 2017-11-14 15:09:38:307 HookProcessor.cpp(261) Entering HookProcessor::PushData
T 2017-11-14 15:09:38:308 HookProcessor.cpp(287) Entering HookProcessor::PullData
D 2017-11-14 15:09:38:308 HookProcessor.cpp(293) Waiting for connection from browser via named pipe
T 2017-11-14 15:09:38:308 CookieManager.cpp(398) Entering CookieManager::ThreadProc
D 2017-11-14 15:09:38:309 HookProcessor.cpp(302) Connection from browser established via named pipe
T 2017-11-14 15:09:38:309 CookieManager.cpp(258) Entering CookieManager::ReadPersistentCookieFile
T 2017-11-14 15:09:38:310 CookieManager.cpp(302) Entering CookieManager::ParsePersistentCookieInfo
```

My suspicion is that this is something introduced by an update to Windows 10, possibly the Fall Creator's Update.  I updated a couple of weeks ago and I don't believe I've run these particular tests locally against IE since then.  My tests are still running fine on our grid which is running nodes on Windows Server 2016 and I haven't run any tests on IE locally that make use of this functionally in a while,

**Comment by barancev** [comment]
Can you plese retest with IEDriverServer version 3.7?

**Comment by bjharo** [comment]
@barancev The same thing happens with version 3.7.

**Comment by hatanaka-akihiro** [comment]
I faced the same problem Selenium 3.8.1 with Java 8.

OS:
Windows 10 Pro (64 bit)
Version 1709
Build: 16299.19

Selenium Version:
3.8.1

Browser:
Internet Explorer

Browser Version:
11.64.16299.0

**Comment by theproshin** [comment]
OS:
Windows 10 Корпоративная (64 bit)
Version 1709
Build: 16299.15

Selenium Version:
3.8.1

Browser:
Internet Explorer

Browser Version:
11.15.16299.0

```
>>> cap = DesiredCapabilities.INTERNETEXPLORER.copy()
>>> driver = webdriver.Remote(desired_capabilities=cap, command_executor='http://localhost:4444/wd/hub')
>>> driver.get('http://ya.ru')
>>> driver.get_cookies()
Traceback (most recent call last):
  File "<input>", line 1, in <module>
  File "C:\VENV\selenium\lib\site-packages\selenium\webdriver\remote\webdriver.py", line 725, in get_cookies
    return self.execute(Command.GET_ALL_COOKIES)['value']
  File "C:\VENV\selenium\lib\site-packages\selenium\webdriver\remote\webdriver.py", line 308, in execute
    self.error_handler.check_response(response)
  File "C:\VENV\selenium\lib\site-packages\selenium\webdriver\remote\errorhandler.py", line 194, in check_response
    raise exception_class(message, screen, stacktrace)
selenium.common.exceptions.WebDriverException: Message: java.net.ConnectException: Connection refused: connect
Stacktrace:
    at org.openqa.selenium.remote.server.WebDriverServlet.lambda$handle$0 (WebDriverServlet.java:245)
    at java.util.concurrent.Executors$RunnableAdapter.call (None:-1)
    at java.util.concurrent.FutureTask.run (None:-1)
    at java.util.concurrent.ThreadPoolExecutor.runWorker (None:-1)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run (None:-1)
    at java.lang.Thread.run (None:-1)

```

**Comment by ppptrevoro** [comment]
Just had fall update and now only iedriverserver.exe is crashing when I call my `Teardown()` method containing `_driver.Manage().Cookies.DeleteAllCookies();` C# Selenium 3.8

```
Faulting application name IEDriverServer.exe, version 3.3.0.0, time stamp 0x58bf0315
Faulting module name IED1106.tmp, version 0.0.0.0, time stamp 0x58bf0313
Exception code 0xc0000005
Fault offset 0x0009d640
Faulting process id 0x1af0
Faulting application start time 0x01d37826f882420a
Faulting application path C:\\TestDrivers\IEDriverServer.exe
Faulting module path C:\\Users\\usr_name\\AppData\\Local\\TempIED1106.tmp
Report Id 14048bb1-4d8a-4348-a5c2-2d34e833ee6c
Faulting package full name 
Faulting package-relative application ID 
```

**Comment by joejoseph00** [comment]
A similar situation with the php web-driver 
To prevent IEDriverServer.exe from crashing and get IE11 to work with seleniun, I had to disable the Delete cookie from this part of the php-webdriver:
   `instaclick/php-webdriver/lib/WebDriver/Session.php`

```
    /**
     * Delete all cookies: /session/:sessionId/cookie (DELETE)
     *
     * @return \WebDriver\Session
     */
    public function deleteAllCookies()
    {
        //$this->curl('DELETE', '/cookie');

        return $this;
    }
```

**Comment by eyal919** [comment]
+1

**Comment by jimevans** [comment]
This should have been fixed in IEDriverServer 3.11.1 or later. Anyone still having this issue will need to update to that version, and if it still reproduces, file a new issue with logs from IEDriverServer.exe.
